compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/irCodegenUtils.kt

```diff
@@ -392,12 +392,16 @@ internal fun classFileContainsMethod(classId: ClassId, function: IrFunction, con
 }
 
 val IrMemberWithContainerSource.parentClassId: ClassId?
-    get() = (containerSource as? JvmPackagePartSource)?.classId ?: (parent as? IrClass)?.classId
+    get() = ((this as? IrSimpleFunction)?.correspondingPropertySymbol?.owner ?: this).let { directMember ->
```

Comment:
Is it possible to add any test here?

Reply:
Done. There already was a boxInline test for JvmPackageName, so I just marked the property in it as `inline` so that the inliner will fail is `parentClassId` is `foo.bar._1Kt` instead of `baz.foo.quux.bar._1Kt` as specified in the annotation.

Reply:
@udalov have some intention for  non-inline property here. Looks like it's best add new inline one and keep old one

Reply:
done